### ISSUE TITLE
MARKs for nested types not possible?

### ISSUE THREAD

**Issue by Sajjon** [nicklockwood/SwiftFormat#1204]
The rule `marktypes` inserts `// MARK: - <TYPE_NAME>` which is excellent!
It also MARKs protocol conformance in extensions with the type and the protocol name, excellent!

However, it does not MARK _**any**_ nested type at all. Am I stupid? Am I missing something? I used Nested types, everywhere! And here swiftformat seems to fall short?

This code

```swift
public enum App {}

public extension App {
	struct State: Equatable {
		public var alert: AlertState<Action>?

		public init(
			alert: AlertState<Action>? = nil,
		) {
			self.alert = alert
		}
	}
}
```

is by swiftformat change to this code:

```swift
// MARK: - App
public enum App {}

public extension App {
	struct State: Equatable {
		public var alert: AlertState<Action>?

		public init(
			alert: AlertState<Action>? = nil,
		) {
			self.alert = alert
		}
	}
}
```

With this `.swiftformat` config file:

```
--swiftversion 5.6

# format options
--allman false
--indent tab
--tabwidth 4

--enable marktypes
--lineaftermarks false

# file options
--exclude .build
```

But what I expected it for it to generate this code:

```swift
// MARK: - App
public enum App {}

// MARK: App.State
public extension App {
	struct State: Equatable {
		public var alert: AlertState<Action>?

		public init(
			alert: AlertState<Action>? = nil,
		) {
			self.alert = alert
		}
	}
}
```

Or possibly like this:

```swift
public extension App {
        // MARK: State
	struct State: Equatable {
```

Is that possible? :)

**Comment by calda** [comment]
I think this is because the `--organizetypes` option exclude `extension`s by default. This is because I think it only makes sense to organize extension bodies when also using the `extensionAccessControl` rule with `--extensionacl on-delcarations` (which is not the default). 

If you add `extension` to your `--organizetypes` option (`--organizetypes class,actor,struct,enum,extension`) then it should start organizing extension bodies, and the types nested within them.

**Comment by calda** [comment]
It would likely make sense to organize the declarations of types in extensions, without also organizing the rest of the extension, but this is not how the rule is currently implemented.

**Comment by calda** [comment]
The `markTypes` rule currently doesn't support marking nested types. It could be reasonable to add a `// MARK: App.State` in your `extension App { struct State { ... } }` example (where the extension contains exactly one declaration, which is a type).

**Comment by calda** [comment]
Thanks for the suggestion @Sajjon! Implemented in #1211.

**Comment by Sajjon** [comment]
Fantastic! Good job and thanks a lot for fast impm!

**Comment by nicklockwood** [comment]
@Sajjon fixed in 0.49.12

**Comment by nicklockwood** [comment]
@Sajjon sorry false alarm, this is still only in develop.

**Comment by nicklockwood** [comment]
@Sajjon landed in 0.50.0
